### PR TITLE
Update vuln query to utilize hasSBOM and pass in SBOM URI or purl to search

### PR DESCRIPTION
**Pull request by pxp928**
# Description of the PR

update vulnerability query to utilize hasSBOM in query and be searchable via sbom URI or purl (this will make searching for packages with guac generated purls easier).

For CDX - the URI is the `"serialNumber": "urn:uuid:bc2d1509-298e-42cd-a4d0-b5fac740b4ae"`
For SPDX - the URI is the `"documentNamespace": "https://anchore.com/syft/image/k8s.gcr.io/kube-apiserver-v1.24.1-583a02ce-8f7e-4794-91af-35f27ffeb73d"` (which is unique per the spec: https://spdx.github.io/spdx-spec/v2.3/document-creation-information/#65-spdx-document-namespace-field)

[guac vuln CLI documentation](https://docs.guac.sh/querying-via-cli/) will have to be updated.

# PR Checklist

- [x] All commits have [a Developer Certificate of Origin (DCO)](https://wiki.linuxfoundation.org/dco) -- they are generated using `-s` flag to `git commit`.
- [ ] All new changes are covered by tests
- [x] If GraphQL schema is changed, `make generate` has been run
- [ ] If `collectsub` protobuf has been changed, `make proto` has been run
- [ ] All CI checks are passing (tests and formatting)
- [ ] All dependent PRs have already been merged

### CODE REVIEW

**Review by mihaimaruseac — APPROVED** [review]

**Review by mihaimaruseac — APPROVED** [review]

**Comment by pxp928** [comment]
Rebased based on https://github.com/guacsec/guac/pull/1540

File: pkg/assembler/graphql/schema/hasSBOM.graphql [hunk 1]
```diff
@@ -61,9 +61,9 @@ input HasSBOMSpec {
   origin: String
   collector: String
   knownSince: Time
-  includedSoftware: [PackageOrArtifactSpec]!
-  includedDependencies: [IsDependencySpec]!
-  includedOccurrences: [IsOccurrenceSpec]!
+  includedSoftware: [PackageOrArtifactSpec]
+  includedDependencies: [IsDependencySpec]
+  includedOccurrences: [IsOccurrenceSpec]
```
Comment (jeffmendoza) [inline pkg/assembler/graphql/schema/hasSBOM.graphql:66]: Should be
```
 includedSoftware: [PackageOrArtifactSpec!]
  includedDependencies: [IsDependencySpec!]
  includedOccurrences: [IsOccurrenceSpec!]
```
right?

File: cmd/guacone/cmd/vulnerability.go [hunk 2]
```diff
@@ -42,18 +43,18 @@ type queryOptions struct {
...
+	searchString    string
+	isPurl          bool
 	vulnerabilityID string
 	depth           int
 	pathsToReturn   int
 }
 
 var queryVulnCmd = &cobra.Command{
-	Use:   "vuln [flags] purl",
+	Use:   "vuln [flags] purl/sbomURI",
```
Comment (jeffmendoza) [inline cmd/guacone/cmd/vulnerability.go:54]: nit:
```
"vuln [flags] <purl/sbomURI>"
```

**Review by jeffmendoza — APPROVED** [review]
Question on the gql change.

File: cmd/guacone/cmd/vulnerability.go (hunk 2 above)
Comment (pxp928) [reply cmd/guacone/cmd/vulnerability.go:54]: fixed

File: pkg/assembler/graphql/schema/hasSBOM.graphql (hunk 1 above)
Comment (pxp928) [reply pkg/assembler/graphql/schema/hasSBOM.graphql:66]: ah yes! Good catch.

**Comment by pxp928** [comment]
Documentation updated: https://github.com/guacsec/guac-docs/pull/113

**Pull request merged**